htdocs/core/lib/functions.lib.php

```diff
@@ -12291,6 +12291,39 @@ function dolGetButtonAction($label, $text = '', $actionType = 'default', $url =
 
 	// If $url is an array, we must build a dropdown button or recursively iterate over each value
 	if (is_array($url)) {
+
+		/**
+		 * An anonymous function to complete dropdown url
+		 * @param string $url
+		 * @return string
+		 */
+		$completeUrl = function ($url) use ($params) {
+			if (empty($url)) {
+				return '';
+			}
+
+			$parsedUrl = parse_url($url);
+			if (isset($parsedUrl['scheme']) && $parsedUrl['scheme'] === 'javascript') {
+				return $url;
+			}
+
+			if (!empty($parsedUrl['query'])) {
+				// Use parse_str() function to parse the string passed via URL
+				parse_str($parsedUrl['query'], $urlQuery);
+				if (!isset($urlQuery['backtopage']) && isset($params['backtopage'])) {
+					$url.= '&amp;backtopage='.urlencode($params['backtopage']);
+				}
+			}
+
+			// TODO : this is a first fix of button behavior : external module will needs to use http:// or https:// prefix on there Urls
+			if (!isset($parsedUrl['scheme'])) {
+				$url = DOL_URL_ROOT.$url;
+			}
+
+			return $url;
+		};
```

Comment:
@eldy I think you've misunderstood what I mean with this function.
In some case if href attribute of link start with javascript: because a link can have differrent sheme like

- http:
- https:
- javascript:
- maito:
- #

but in all cases the code force to add `&backtopage=` to url. Also  the actual dolibarr code doesn't check if query already contain  backtopage param.

I will recreate 2 PRs 
- first to fix ajax loaded BTN
- second to allow desabling `&backtopage=`